docs/source/yaml_basics.rst

```diff
@@ -0,0 +1,122 @@
+model configurations but do not go in-depth about that topic. For more information about model configurations, please
+see that section of this documentation.
+
+Structure
+---------
+
+YAML files are structured by lines and space indentations. Indentation levels should be either 2 or 4 spaces, and
+**tabs are not valid**.  For example, a configuration file that includes a ``BasePopulation()`` component would look
+like the following:
+
+.. code-block:: yaml
+
+    components:
+        vivarium_public_health:
+            population:
+                - BasePopulation()
+
+Comments
+--------
+
+YAML comments are denoted with the pound symbol ``#``, and can be placed anywhere, but must be separated from the
+preceding token by a space. For example, adding a comment to the configuration from above looks like this:
+
+.. code-block:: yaml
+
+    components:
+        vivarium_public_health:
+            population:
+                - BasePopulation()  # Produces and ages simulants, no death
+
+
+Mappings
+--------
+
+A mapping, or key-value pairing, is formed using a colon `:`. This corresponds to an entry from the ``dictionary``
+data structure from python, and there is no notion of ordering. Mappings can be specified in block format or inline,
+however we recommend block format so that is what we will show an example of here. In block format, mappings are
+separated onto new lines, and indentation forms a parent-child relationship. For example, below is a snippet from a
+configuration that specifies configuration parameters for a simulation population as mappings. Each colon below begins
+a mapping.
+
+.. code-block:: yaml
+
+    configuration:
+        population:
+            population_size: 1000
+            age_start: 0
+            age_end: 30
+
+The interpretation of this configuration into python is shown below . You may have noticed that the above example
+contains nested mappings, this is valid YAML syntax and it relies on whitespace indentation. Also, the inner most block
+(population_size, age_start, age_end) is unordered.
+
+.. code-block:: python
+
+    {configuration: {
+        population: {
+            population_size : 1000,
+            age_start: 0,
+            age_end: 30
+            }
+        }
+    }
+
+Lists
+-----
+
+A list is formed using a hyphen ``-`` in block format, with each entry appearing on a new line with the same indentation
+level.  As with mappings, lists can be nested, and mappings and lists can be intermixed. Below is a configuration
+snippet that specifies a list of components to be used from the ``population`` module in Vivarium Public Health.
+
+.. code-block:: yaml
+
+    components:
+        vivarium_public_health:
+            population:
+                - BasePopulation()
+                - Mortality()
+                - FertilityCrudeBirthRate()
+
+This will be interpreted as
+
+.. code-block:: python
+
+    {components: {
+            vivarium_public_health: {
```

Comment:
e.g. this is not at all how the framework interprets this block.  it gets interpreted as 
```
[vivarium_public_health.population.BasePopulation(), vivarium_public_health.population.Mortality(), vivarium_public_health.population.FertilityCrudeBirthRate()]
```